lib/src/backend.rs

```diff
@@ -201,6 +202,33 @@ impl MergedTreeId {
             MergedTreeId::Merge(tree_ids) => tree_ids.clone(),
         }
     }
+
+    /// Represent a `MergeTreeId` in a user friendly way. This makes no
+    /// stability guarantee, as the format may change at any time.
+    #[allow(clippy::inherent_to_string)]
+    pub fn to_string(&self) -> String {
+        match self {
+            MergedTreeId::Legacy(tree_id) => tree_id.hex(),
+            MergedTreeId::Merge(tree_ids) => {
+                let ids = tree_ids
+                    .map(|id| id.hex())
```

Comment:
match on `self.to_merge().as_resolved()` instead?

`MergedTreeId::Merge(_)` doesn't mean the tree has conflicts, and is equivalent to `Legacy(tree_id)` if `tree_ids.len() == 1`.

Reply:
> match on `self.to_merge().as_resolved()` instead?

Why does the tree have to be resolved for a working-copy name? 

> `MergedTreeId::Merge(_)` doesn't mean the tree has conflicts, and is equivalent to `Legacy(tree_id)` if `tree_ids.len() == 1`.

Makes sense.

Reply:
> > match on `self.to_merge().as_resolved()` instead?
> 
> Why does the tree have to be resolved for a working-copy name?

It doesn't have to be. I just thought you would want just `{id}` (without `+` prefix) in that case.